packages/client/src/utils/promises.ts

```diff
@@ -270,3 +270,21 @@ export const withThrottling = (fn: (...args: any[]) => Promise<any>, maxInvocati
     })
     return throttler(fn)
 }
+
+export const tryInSequence = async <T>(fns: ((...args: any[]) => Promise<T>)[]): Promise<T | never> => {
+    if (fns.length === 0) {
+        throw new Error('no tasks')
+    }
+    let firstError: any
+    for (const fn of fns) {
+        try {
+            const promise = fn()
+            return await promise
+        } catch (e: any) {
+            if (firstError === undefined) {
+                firstError = e
```

Comment:
Wondering if it could be useful to report all of the errors for users. Although, most of the time if all of the requests fail it's likely due to the same reason